### PR TITLE
Add a GET endpoint for labels

### PR DESCRIPTION
**Pull request by geoffjentry**

### CODE REVIEW

**Comment by geoffjentry** [comment]
@kshakir I wasn't sure how the sbt magic you put in for the API docs worked, so I probably didn't do whatever was necessary

**Comment by kshakir** [comment]
> I wasn't sure how the sbt magic you put in for the API docs worked

Pass the word on. Currently available both in the updated [wiki](https://github.com/broadinstitute/cromwell/wiki/DevZone#generating-a-markdown-document-of-the-swagger-yaml) and the more comprehensive mega doc-on-docs in our [team drive](https://drive.google.com/drive/u/1/folders/0By3wA7o30lk3VmF3aldBNkEzaDg).

File: engine/src/main/resources/swagger/cromwell.yaml [hunk 1]
```diff
@@ -140,6 +140,27 @@ paths:
         - google_oauth:
             - openid
   '/api/workflows/{version}/{id}/labels':
+    get:
+      summary: Retrieves the current labels for a workflow
+      parameters:
+        - $ref: '#/parameters/versionParam'
+        - $ref: '#/parameters/singleId'
```
Comment (cjllanwarne) [inline engine/src/main/resources/swagger/cromwell.yaml]: ToL: is `singleId` the same name used by other lookup endpoints? I've not noticed it before but it's quite possible that it is!

Comment (cjllanwarne) [reply engine/src/main/resources/swagger/cromwell.yaml]: But if it is, change the URL format which is currently `id`

File: services/src/main/scala/cromwell/services/metadata/MetadataService.scala [hunk 2]
```diff
@@ -65,15 +65,16 @@ object MetadataService {
...
-  case class GetStatus(workflowId: WorkflowId) extends ReadAction
-  case class WorkflowQuery(parameters: Seq[(String, String)]) extends ReadAction
-  case class WorkflowOutputs(workflowId: WorkflowId) extends ReadAction
-  case class GetLogs(workflowId: WorkflowId) extends ReadAction
+  final case class GetMetadataQueryAction(key: MetadataQuery) extends ReadAction
+  final case class GetStatus(workflowId: WorkflowId) extends ReadAction
+  final case class GetLabels(workflowId: WorkflowId) extends ReadAction
+  final case class WorkflowQuery(parameters: Seq[(String, String)]) extends ReadAction
+  final case class WorkflowOutputs(workflowId: WorkflowId) extends ReadAction
+  final case class GetLogs(workflowId: WorkflowId) extends ReadAction
```
Comment (cjllanwarne) [inline services/src/main/scala/cromwell/services/metadata/MetadataService.scala]: Nice to see somebody's `final`ly done this!

File: engine/src/main/resources/swagger/cromwell.yaml (hunk 1 above)
Comment (geoffjentry) [reply engine/src/main/resources/swagger/cromwell.yaml]: This was a copy/paste w/ changes from some other entry, so at least one does. I'm just lookingat the thumbnail but I think `singleId` is just a swagger alias

Comment (geoffjentry) [reply engine/src/main/resources/swagger/cromwell.yaml]: Confirming thisi s just an artifact of swagger yaml

Comment (cjllanwarne) [reply engine/src/main/resources/swagger/cromwell.yaml]: Just cross reference against the URL about 5 lines up and 👍

**Comment by geoffjentry** [comment]
@kshakir The pieces I put in are as well tested as their siblings and they'll remain that way as this PR, so Codecov can remain unhappy.

**Comment by geoffjentry** [comment]
on the failures - the extra commit at the tail end (cromwell api) i threw in quickly last night after i submitted the PR w/ the intention of finishing it this morning. apparently it didn't even compile. Busted!

File: cromwellApiClient/src/main/scala/cromwell/api/model/WorkflowLabels.scala [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package cromwell.api.model
+
+import spray.json.{DefaultJsonProtocol, JsObject, JsValue}
+
+object WorkflowLabelsJsonSupport extends DefaultJsonProtocol {
+  implicit val LabelsResponseFormat = jsonFormat2(WorkflowLabels)
+}
+
+final case class WorkflowLabels(id: String, labels: JsObject)
```
[the remark this answers is not in the document]
Comment (geoffjentry) [reply cromwellApiClient/src/main/scala/cromwell/api/model/WorkflowLabels.scala]: fixed

File: docs/api/RESTAPI.md [hunk 4]
```diff
@@ -57,7 +57,7 @@ Submits a workflow to Cromwell. Note that this endpoint can accept an unlimited
 |**FormData**|**workflowInputs_5**  <br>*optional*|A fifth JSON or YAML file containing inputs.|file||
 |**FormData**|**workflowOptions**  <br>*optional*|JSON file containing configuration options for the execution of this workflow.|file||
 |**FormData**|**workflowSource**  <br>*required*|The workflow source file to submit for execution.|file||
-|**FormData**|**workflowType**  <br>*optional*|The workflow language for the file you submitted. Cromwell currently supports WDL.|enum (wdl, cwl)|`"wdl"`|
+|**FormData**|**workflowType**  <br>*optional*|The workflow language for the file you submitted. Cromwell currently supports WDL.|string||
```
Comment (geoffjentry) [inline docs/api/RESTAPI.md]: looks like some stowaways from other people's doc-changing events

File: engine/src/main/resources/swagger/cromwell.yaml (hunk 1 above)
Comment (geoffjentry) [reply engine/src/main/resources/swagger/cromwell.yaml]: @cjllanwarne I'm not sure what you're asking and/or saying. Look at the abort/status/logs/etc endpoints, this is the same

File: docs/api/RESTAPI.md (hunk 4 above)
Comment (kshakir) [reply docs/api/RESTAPI.md]: ToL: I'm fine leaving this non-automated as long as reviewers are checking on each other / it doesn't really matter when we re-run the generator. If we want to make this even more automated in some way, that's always an option too. Alternatively, we could also have folks quickly scan a PR [checklist](https://www.npr.org/2017/10/30/559996276/the-trick-to-surviving-a-high-stakes-high-pressure-job-try-a-checklist).

Comment (geoffjentry) [reply docs/api/RESTAPI.md]: I'd love to see it automated (I was kind of hoping it already was) and personally woudln't mind a checklist, but checklists have been soundly rejected whenever they've come up

**Pull request merged**